core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -200,21 +204,21 @@ protected void add(DataFile file) {
    * Add a delete file to the new snapshot.
    */
   protected void add(DeleteFile file) {
-    setWriteSpec(file);
-    addedFilesSummary.addedFile(writeSpec(), file);
+    Preconditions.checkNotNull(file, "Invalid delete file: null");
+    PartitionSpec fileSpec = ops.current().spec(file.specId());
+    List<DeleteFile> deleteFiles = newDeleteFilesBySpec.computeIfAbsent(file.specId(), specId -> Lists.newArrayList());
+    deleteFiles.add(file);
```

Comment:
Hello @aokolnychyi, may I ask, what is `deleteFiles`'s role here? It seems like it can be safely deleted?